### DISCUSSION TITLE
Local Backend without Netlify

### DISCUSSION THREAD

**Discussion by mnordhaus** [StaticJsCMS/static-cms#986] [Q&A] [answered]
Hi,

first of all thank you for maintaining this great peace of software!

I try to run the CMS locally with my local git repo using local_backend: true 

The CMS tries to authenticate through netlify tho and asks for the URL of the netlify site.

<img width="618" alt="image" src="https://github.com/StaticJsCMS/static-cms/assets/1510804/a219657b-0d6b-4656-9390-d14b22fb123c">

<img width="652" alt="image" src="https://github.com/StaticJsCMS/static-cms/assets/1510804/ec0f5434-6ada-4330-b6a3-47ec09642a7b">

Is it possible to deactivate this behavior? My expectation would be that the CMS would open without talking to netlify at all and just do its thing locally. 

Thanks for all help! The CMS is really great and has huge potential to get big!

Best,

Markus

**Comment by mnordhaus** [comment]
After entering the Netlify site URL I was able to "log in" but now my local netlify commits to the github repo and not the local git.

My Backend Config:

backend:
  name: git-gateway
  branch: main
  local_backend: true

And I have the proxy is running:

![image](https://github.com/StaticJsCMS/static-cms/assets/1510804/2a137b3d-8cf8-4eab-bc0b-ad794abac571)

**Comment by mnordhaus** [comment]
accidentally closed it any help is aprecciated ...

**Comment by mnordhaus** [comment]
And this is my CMS Page using Hugo:

`<!DOCTYPE html>
<html>
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <link rel="stylesheet" href="https://unpkg.com/@staticcms/app@^3.4.5/dist/main.css" />
    
    <title>Content Manager</title>
  </head>
  <body>
    
    <script src="https://unpkg.com/@staticcms/app@^3.4.5/dist/static-cms-app.js"></script>
    <script>
      window.CMS.init();
    </script>
  </body>
</html>`

As you see I incommented the Netflify widget. I did that as well in the header for the rest of my pages. But still the same result

**Reply by mnordhaus** [reply]
Well I get this screen now:

<img width="581" alt="image" src="https://github.com/StaticJsCMS/static-cms/assets/1510804/9b936fa3-72f5-4055-a431-27b27cd429c4">

**Comment by denyskon** [comment]
You need the proxy server to use local backend, take a look at https://www.staticcms.org/docs/local-backend

**Reply by mnordhaus** [reply]
Thank you @denyskon for your reply. But is the whole point of proxy server not exactly for that? I mean it is the local backend.

The summary of my config is written done here: https://github.com/StaticJsCMS/static-cms/discussions/986#discussioncomment-7591985

And you see in the screenshot that the proxy server is running on the default port 8081. I do not see any configuration options for the proxy server except of setting a different port.

**Reply by denyskon** [reply]
If everything is running correctly it should work without Netlify when the CMS detects a local backend.... seems like a big then if it isn't a misconfiguration

**Reply by mnordhaus** [reply]
Thank you, that is what I am thinking now too. But I did not want to be that guy that opens a ticket that is actually an issue sitting in front of the screen :-).

**Comment by mnordhaus** [comment]
Created an issue: https://github.com/StaticJsCMS/static-cms/issues/989#issue-1998345545 and will update this discussion accordingly.

**Comment by denyskon** [comment]
@mnordhaus I think local_backend: true should be top level, not inside `backend`

**Reply by mnordhaus** [reply, marked as answer]
@denyskon That made my local development work. I also opened a pull request for an improvement of the documentation. I personally have overseen the indentation. I assume this little tweak might help:

https://github.com/StaticJsCMS/static-cms/pull/993#issue-2000420652

Thanks @denyskon
